src/model/parameter.cpp

```diff
@@ -0,0 +1,39 @@
+#include <deepworks/parameter.hpp>
+#include <deepworks/tensor.hpp>
+
+struct deepworks::Parameter::Impl {
+    deepworks::Tensor m_data;
+    deepworks::Tensor m_grad;
+    bool              m_is_trainable;
+};
+
+deepworks::Parameter::Parameter(deepworks::Tensor&& data, bool is_trainable)
+    : m_impl(new deepworks::Parameter::Impl()) {
+    m_impl->m_data = std::move(data);
+    m_impl->m_grad = deepworks::Tensor(m_impl->m_data.shape());
+    m_impl->m_is_trainable = is_trainable;
+}
+
+const deepworks::Tensor& deepworks::Parameter::data() const {
+    return m_impl->m_data;
+}
+
+deepworks::Tensor& deepworks::Parameter::data() {
+    return m_impl->m_data;
+}
+
+const deepworks::Tensor& deepworks::Parameter::grad() const {
+    return m_impl->m_grad;
+}
+
+deepworks::Tensor& deepworks::Parameter::grad() {
+    return m_impl->m_grad;
+}
+
+void deepworks::Parameter::train(bool mode) {
```

Comment:
Should we rename this method to set_mode?
Or we can have eval() and train() methods to switch the mode.

Reply:
I believe `set_mode` can be used directly for entire `dw::Model`, when this one used only for freeze specific `param`